src/main/java/seedu/duke/Duke.java

```diff
@@ -3,19 +3,116 @@
 import java.util.Scanner;
 
 public class Duke {
-    /**
-     * Main entry-point for the java.duke.Duke application.
-     */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-        System.out.println("What is your name?");
-
+        ExpenseTracker expenseTracker = new ExpenseTracker();
         Scanner in = new Scanner(System.in);
-        System.out.println("Hello " + in.nextLine());
+        System.out.println("Hiya! How can I assist?");
+        while (true) {
+            if (in.hasNextLine()) {
+                String input = in.nextLine().trim();
+                if (input.startsWith("add-expense")) {
+                    addExpenseRequest(input, expenseTracker);
+                } else if (input.startsWith("add-category")) {
+                    expenseTracker.addCategory(input);
+                }else if (input.startsWith("delete-expense")) {
+                    deleteExpenseRequest(input, expenseTracker);
+                } else if (input.startsWith("tag-expense")) {
+                    expenseTracker.tagExpense(input);
+                } else if (input.equalsIgnoreCase("bye")) {
+                    System.out.println("Goodbye! :> Hope to see you again soon!");
+                    break;
+                } else if (input.equalsIgnoreCase("view-budget")) {
+                    expenseTracker.viewBudget();
+                } else if (input.startsWith("set-budget")) {
+                    setBudgetLimitRequest(input, expenseTracker);
+                } else if (input.startsWith("view-expenses")) {
+                    expenseTracker.viewExpensesByCategory();
+                } else if (input.equalsIgnoreCase("toggle-reset")) {
+                    expenseTracker.toggleAutoReset();
+                } else {
+                    System.out.println("Invalid input! Try again."); // Provide feedback for invalid input
+                }
+            } else {
+                System.out.println("No input received.");
+                break;
+            }
+        }
+    }
+
+    public static void addExpenseRequest(String input, ExpenseTracker expenseTracker) {
+        try {
+            String[] parts = input.split(" ");
+
+            String name = null;
+            double amount = 0;
+            String category = null;
+
+
```

Comment:
Consider to add a check that ensures amounts are not negative in addExpense and setBudgetLimit.